### PR TITLE
fix: Latest comment admin imports

### PR DESCRIPTION
**Pull request by tofumatt**
This is a fix to the latest comments block from #7941; apparently importing that admin file didn't work super-well and caused errors when re-generating fixtures, etc.

The function we're using is trivial to re-implement and I think the best solution for now. It also let me regenerate the fixtures.

### CODE REVIEW

File: core-blocks/latest-comments/index.php [hunk 1]
```diff
@@ -5,9 +5,30 @@
...
+ *
+ * @since 3.3.0
+ *
+ * @param int|WP_Post $post Optional. Post ID or WP_Post object. Default is global $post.
+ * @return string The post title if set; "(no title)" if no title is set.
+ */
+function gutenberg_draft_or_post_title( $post = 0 ) {
+	$title = get_the_title( $post );
+	if ( empty( $title ) ) {
+		$title = __( '(no title)' );
```
Comment (Soean) [inline core-blocks/latest-comments/index.php]: Text domain is missing

Comment (tofumatt) [reply core-blocks/latest-comments/index.php]: Should be `__( '(no title)', 'gutenberg' );`, right? 😄

**Comment by pento** [comment]
@tofumatt: Is there any particular reason for using `_draft_or_post_title()`, apart from the obvious?

It seems that Core has had the behaviour of not displaying a title since forever, with no bug reports, which kind of makes sense: if a CPT doesn't allow a `post_title`, there's a good chance it probably doesn't do comments and the Recent Comment widget, either.

**Comment by tofumatt** [comment]
The idea here is the comments look very strange if you don’t have a post title. I suppose we could add logic to just show the comment. But then it would just be the commenter’s name, the date, and the comment (if enabled).

Having “No title” at least gives the comment a link to that post. "Recent posts” in core’s default theme do what I’m doing here, so it seems better to say “No title” than to show nothing (which might read more like a bug).

**Comment by aduth** [comment]
It can be done separately, but I found we could have slightly-improved support for WordPress functions via the `SHORTINIT` constant, which could make some of our ad-hoc `require` in `bin/get-server-blocks.php` redundant (i.e. `require ABSPATH . 'wp-settings.php';`). 

https://github.com/WordPress/wordpress-develop/blob/796bac86451a4fb626b298181c9ac1f37bc6c441/src/wp-settings.php#L133-L136

(Still wouldn't have fixed the original issue on its own, though)

**Comment by aduth** [comment]
In general, I think we should want to encourage avoiding dependency on `wp-admin` files within block implementations (since they're used in non-wp-admin contexts).

File: core-blocks/latest-comments/index.php [hunk 2]
```diff
@@ -5,9 +5,30 @@
...
+/**
+ * Get the post title.
+ *
+ * The post title is fetched and if it is blank then a default string is
+ * returned.
+ *
+ * Copied from `wp-admin/includes/template.php`, but we can't include that
+ * because:
+ *
+ * 1. it causes bugs
```
Comment (aduth) [inline core-blocks/latest-comments/index.php]: Could be more specific here.

File: core-blocks/latest-comments/index.php [hunk 3]
```diff
@@ -5,9 +5,30 @@
...
+ * 2. it's in the admin; ideally we *shouldn't*
```
Comment (aduth) [inline core-blocks/latest-comments/index.php]: Incomplete sentence?

**Review by aduth — APPROVED** [review]

**Pull request merged**